lib/msal-common/src/request/AuthenticationHeaderParser.ts

```diff
@@ -54,20 +55,40 @@ export class AuthenticationHeaderParser {
         throw ClientConfigurationError.createMissingNonceAuthenticationHeadersError();
     }
 
+    /**
+     * This method parses the claims value out of the WWW-Authenticate authentication header.
+     * See: https://docs.microsoft.com/en-us/azure/active-directory/develop/claims-challenge
+     * @returns 
+     */
+    getClaims(): string {
+        // Attempt to parse claims from WWW-Authenticate
+        const wwwAuthenticate = this.headers[HeaderNames.WWWAuthenticate];
+        if (wwwAuthenticate) {
+            const wwwAuthenticateChallenges = this.parseChallenges<WWWAuthenticateChallenges>(wwwAuthenticate);     
+            if (wwwAuthenticateChallenges.claims){
+                return wwwAuthenticateChallenges.claims;
+            }
+            throw ClientConfigurationError.createInvalidAuthenticationHeaderError(HeaderNames.WWWAuthenticate, "claims challenge is missing.");
```

Comment:
To me it seems strange to throw exceptions. Exceptions means some error happened. But not having claims in a WWW-Authenticate header is normal. In fact in most cases there won't be any claims.

Reply:
@bgavrilMS I would say in this particular context it makes sense, since as a dev I'm expecting it to return a claims string and would probably appreciate knowing it didn't do so before passing it to where I will need it (i.e. `acquireToken*`). Returning null or empty string could easily get overlooked since `acquireToken` will not warn down the line. There's also the matter of consistency with `getShrNonce`. Open to alternatives though: what would be a good way to notify the dev?

```
try {
   const authHeaderParser = new AuthenticationHeaderParser(response.headers);
   const claimsChallenge = authHeaderParser.getClaims(); 
   // ...
} catch {
  // ...
}
```

Reply:
Ah, I see. Should definitely try to align this with the .NET implementation then. Will update.